experimental/Schemes/IdealSheaves.jl

```diff
@@ -0,0 +1,470 @@
+      R == ambient_ring(U) && return IdealSheaf(X, C, U, g)
+    end
+  end
+  error("the given set of generators could not be associated to an affine patch of the scheme")
+end
+
+function IdealSheaf(X::CoveredScheme, C::Covering, g::Vector{RET}) where {RET<:MPolyElem} 
+  length(g) == 0 && IdealSheaf(X)
+  R = parent(g[1])
+  for i in 2:length(g)
+    R == parent(g[i]) || error("elements do not belong to the same ring")
+  end
+  C in coverings(X) || error("covering is not listed")
+  for U in patches(C)
+    R == base_ring(OO(U)) && return IdealSheaf(X, C, U, g)
+  end
+  error("the given set of generators could not be associated to an affine patch of the scheme")
+end
+
+# pullback of an ideal sheaf for internal use between coverings of the same scheme
+function (F::CoveringMorphism)(I::IdealSheaf)
+  X = scheme(I)
+  D = codomain(F)
+  D == covering(I) || error("ideal sheaf is not defined on the correct covering")
+  C = domain(F)
+  new_dict = Dict{AbsSpec, Ideal}()
+
+  # go through the patches of C and pull back the generators 
+  # whenever they are defined on the target patch
+  for U in patches(C)
+    f = F[U]
+    V = codomain(f)
+    # for the basic patches here
+    if haskey(ideal_dict(I), V)
+      new_dict[U] = ideal(OO(U), pullback(f).(I[V]))
+    end
+    # check for affine refinements
+    if haskey(affine_refinements(D), V)
+      Vrefs = affine_refinements(D)[V]
+      # pull back the refinement
+      for W in Vrefs
+        h = pullback(f).(gens(W))
+        # take care to discard possibly empty preimages of patches
+        j = [i for i in 1:length(h) if !iszero(h)]
+        Wpre = SpecOpen(U, h[j])
+        add_affine_refinement!(C, Wpre)
+        for i in 1:length(j)
+          if haskey(ideal_dict(I), Wpre[i])
+            new_dict[Wpre[i]] = lifted_numerator.(pullback(f).(I[V[j[i]]]))
+          end
+        end
+      end
+    end
+  end
+  return IdealSheaf(X, C, new_dict)
+end
+
+function +(I::IdealSheaf, J::IdealSheaf) 
+  X = scheme(I)
+  X == scheme(J) || error("ideal sheaves are not defined over the same scheme")
+  new_dict = Dict{AbsSpec, Ideal}()
+  CI = covering(I)
+  CJ = covering(J)
+  CI == CJ || error("addition on ideal sheaves on different coverings is not implemented")
+  for U in patches(CI)
+    new_dict[U] = I[U] + J[U]
+  end
+  return IdealSheaf(X, CI, new_dict, check=false)
+end
+
+function *(I::IdealSheaf, J::IdealSheaf) 
+  X = scheme(I)
+  X == scheme(J) || error("ideal sheaves are not defined over the same scheme")
+  new_dict = Dict{AbsSpec, Ideal}()
+  CI = covering(I)
+  CJ = covering(J)
+  CI == CJ || error("addition on ideal sheaves on different coverings is not implemented")
+  for U in patches(CI)
+    new_dict[U] = I[U] * J[U]
+  end
+  return IdealSheaf(X, CI, new_dict, check=false)
+end
+
+@Markdown.doc """
+    simplify!(I::IdealSheaf)
+
+Replaces the set of generators of the ideal sheaf by a minimal 
+set of random linear combinations in every affine patch. 
```

Comment:
```suggestion
set of random linear combinations of the generators in every affine patch. 
```
How is that a simplification?